### ISSUE TITLE
Metadata format - new (hopefully standardized) fields

### ISSUE THREAD

**Issue by Baklap4** [UltraStar-Deluxe/USDX#714]
### Current metadata

Current metadata tags support common stuff like `ARTIST`, `TITLE`, etc
I'd love to see a couple of new fields added to identify the source of the respective song you're playing.

I have a couple of databases loaded: USDB, ultrastar-es, and some personal songs

and what i'd like to achieve is see where the song came from or which program did create the song. As Ultrastar specifies the format and other programs like vocaluxe and performous follow i'd like to see a / or a couple standardized fields for this in the format.

Personally i'd say:
```
#SOURCE:USDB-Syncer
#SOURCE_VERSION:0.1.0
```

`SOURCE` can be either a website like https://ultrastar-es.org` or an application like `YASS` song creator
For song-creator-applications we might also wanna add a 3rd field... so Creators can opt in aswell

**Comment by Baklap4** [comment]
For starters just discuss the format with the Ultrastar team and see if it'd be possible.
If we then can agree on the different fields (naming etc)
UltraStar Deluxe
Ultrastar Play
Vocaluxe
Performous
And other karaoke programs can use these fields to show where the txt file originated from.

My usecase in this would be to determine if a song came from usdb through the tool usdb-syncer. I recently imported their whole library but some songs were not timed correctly as it tries to guess which audio file needs to be retrieved from youtube
This can cause missmatches and then i can say to the people singing that this song might not work because it was an imported one and not manually checked.

**Comment by Baklap4** [comment]
Sorry for not responding sooner. Some personal issues popped up and i couldn't get to this sooner.

Hmm i see i might have brought in some confusion. What i'd like to achieve is that Ultrastar (originators of the song text format) is considering new fields to be added.
This ticket is not persé in the right place, but i didn't know where to post this elsewhere. If there's a specific repository containing rules/definitions about the songformat which _everyone_ derives from please redirect me there.

What i'd like to achieve is to have a couple more standardized fields besides the already existing ones. Other programs, like usdx, vocaluxe, performous, and creator tools like composer, karedi, yass, etc can then opt in to auto fill-in these fields. Creator tools would use the `#APP` and `#APP_VERSION` to add their name + version whilst websites likes usdb, ultrastar-es and other sources can use the `#SOURCE` field to define where the txt came from as a source of downloadable content.

Defining these fields would make for better automation and maintainability of large libraries where one is constantly updating to the latest version instead of manually adding these things every time an update of the specific txt file is pushed.

The actual karaoke programs can then use this info to present them to the end-user.
In my specific case i'd show it where i can add songs to a playlist. So if i have multiple versions i know in which favor to add them, as quality from the different sites differ a lot...

It's very true that i can add these tags myself to existing txt files. Just `sed` away and it's done. But i can imagine more people are running into this 'problem' and a standardized way is an easy solution for it.

**Comment by dgruss** [comment]
@Baklap4 this would be the repo for the format: https://github.com/UltraStar-Deluxe/format/issues

these fields would be more relevant to song libraries and syncing tools not the game itself

**Comment by barbeque-squared** [comment]
addendum because idk how I missed the root cause instead of the symptom 2 years ago:

> I recently imported their whole library but some songs were not timed correctly
> if i have multiple versions

yeah this is why I vet _every single song_ manually, and if say the USDB version is objectively _better_ than whatever I currently have, whatever I currently have gets nuked. maybe there's a point if contributing back is the goal (though imo that's not something the program itself should concern itself with, because everyone does this differently) but "not timed correctly" is also not the issue with at least 95% of USDB anyway.

EDIT: accidentally hit "close with comment" instead of simply "comment"

**Comment by Baklap4** [comment]
> [@Baklap4](https://github.com/Baklap4) this would be the repo for the format: https://github.com/UltraStar-Deluxe/format/issues
> 
> these fields would be more relevant to song libraries and syncing tools not the game itself

Agreed -> But at that point the format project wasn't there yet or just barely new. This can be closed however since it's already specified within the spec on how to go about this.


> yeah this is why I vet every single song manually, and if say the USDB version is objectively better than whatever I currently have, whatever I currently have gets nuked. maybe there's a point if contributing back is the goal (though imo that's not something the program itself should concern itself with, because everyone does this differently) but "not timed correctly" is also not the issue with at least 95% of USDB anyway.

Also agreed. This was at a time the usdb-syncer itself was just getting started. Since then a lot of things have been changed for the better. And the quality of 'not timed correctly' is almost never the case anymore.


Let's close this issue as it's not relevant anymore
